pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
Agreed about the nervourness, but the watch window error should return `apierrs.FromObject(event.Object)`